targets/TARGET_TOSHIBA/TARGET_TMPM46B/Periph_Driver/src/tmpm46b_esg.c

```diff
@@ -0,0 +1,314 @@
+  */
+uint32_t ESG_GetLatchTiming(void)
+{
+    uint32_t tmp;
+
+    tmp = (TSB_ESG->OUTCR & ESG_LATCHTIMING_SET) >> 16;
+
+    return tmp;
+}
+
+/** 
+  * @brief  Set Entropy seed output timing.
+  * @param  Fintming: the value of entropy seed output timing
+  * @retval The value returned can be one of the following values:
+  *   SUCCESS or ERROR
+  * @register The used registers:
+  *   ESGOUTCR<FINTIMING[15:0]> 
+  */
+Result ESG_SetFintiming(uint16_t Fintming)
+{
+    /* Read OUTCR, keep bit 19 to 16, clear bit 15 to 0 */
+    uint32_t tmp = TSB_ESG->OUTCR & ESG_FINTIMING_CLEAR;
+    Result retval = ERROR;
+
+    /* Get latchtiming value */
+    uint16_t latchtiming = (uint16_t) ESG_GetLatchTiming();
+
+    if (ESG_GetCalculationStatus() == ESG_CALCULATION_COMPLETE) {
+        if (Fintming >= 512U * (latchtiming + 1U) + 3U) {
+            tmp |= (uint32_t) Fintming;
+            TSB_ESG->OUTCR = tmp;
+            retval = SUCCESS;
+        } else {
+            /* Do nothing */
+        }
+    } else {
+        /* Do nothing */
+    }
+
+    return retval;
+}
+
+/** 
+  * @brief  Get entropy seed  Fintiming.
+  * @param  None
+  * @retval tmp: the value of entropy seed Fintiming
+  * @register The used registers:
+  *   ESGOUTCR<FINTIMING[15:0]> 
+  */
+uint16_t ESG_GetFintiming(void)
+{
+    uint16_t tmp;
+
+    tmp = (uint16_t) TSB_ESG->OUTCR & ESG_FINTIMING_SET;
+
+    return tmp;
+}
+
+/** 
+  * @brief  Clear the ESG interrupt.
+  * @param  None
+  * @retval The value returned can be one of the following values:
+  *   SUCCESS or ERROR
+  * @register The used registers:
+  *   ESGINT<INT> 
+  */
+Result ESG_ClrInt(void)
+{
+    Result retval = ERROR;
+
+    if (ESG_GetCalculationStatus() == ESG_CALCULATION_COMPLETE) {
+        /* Write '1' to ESGINT<INT> to clear the interrupt */
+        TSB_ESG->INT = ESG_INT_SET;
+        retval = SUCCESS;
+    } else {
+        /* Do nothing */
+    }
+
+    return retval;
+}
+
+/** 
+  * @brief  Get the ESG interrupt status.
+  * @param  None
+  * @retval The value returned can be one of the following values:
+  *   ENABLE: interrupt occurs 
+  *   DISABLE: no interrupt
+  * @register The used registers:
+  *   ESGINT<INT> 
+  */
+FunctionalState ESG_GetIntStatus(void)
+{
+    FunctionalState retval = DISABLE;
+
+    if (TSB_ESG->INT == ESG_INT_SET) {
+        retval = ENABLE;
+    } else {
+        /* Do nothing */
+    }
+
+    return retval;
+}
+
+/**
+  * @brief  Reset ESG by peripheral function.
+  * @param  None
+  * @retval None
+  * @register The used register:
+  *   SRSTIPRST<IPRST3>
+  *   SRSTPROTECT
+  */
+void ESG_IPReset(void)
+{
+    uint32_t iprst3 = 0U;
+
+    /* Disable write protection state of SRSTIPRST */
+    TSB_SRST->PROTECT = SRST_PROTECT_DISABLE;
+
+    TSB_SRST->IPRST |= SRST_IPRST_3_SET;
+
+    /* Confirm SRSTIPRST<IPRST3> = 1 */
+    do {
+        iprst3 = TSB_SRST->IPRST & SRST_IPRST_3_SET;
+    }
+    while (iprst3 != SRST_IPRST_3_SET);
```

Comment:
This is CPU intensive. Could there be a delay with typical reset time.

Reply:
As this is 3rd party driver (SDK in this case), not a blocker.

What is outstanding in this PR

Reply:
> `This is CPU intensive. Could there be a delay with typical reset time.`

@0xc0170 @mazimkhan, addressed this comment, with the updated driver received from Toshiba Japan.